### DISCUSSION TITLE
custom author field in citar-templates

### DISCUSSION THREAD

**Discussion by wsavran** [emacs-citar/citar#585] [Q&A] [answered]
Hi! I just started using `citar` and I'm really loving the package! I'd like to display a custom author field in my `citar-templates`. I'd like to display the output from `citar-shorten-names` if there are one or two authors, and then {first-author} et al. if there are more than two authors. I'm new to emacs/elisp, so I'm a bit stuck on how to make this happen. Is there a way to do this using existing functions in `citar`? I've looked through citar.el, but i don't see any obvious solutions. Thanks!

**Comment by bdarcus** [comment]
The solution would be to write a new function, similar to `citar-shorten-names`, and to use that instead in the `citar-display-transform-functions`.

If I find time, I'll try to do that and update this and put it on the wiki.

Or, of course, feel free to play around with it; it's the way to learn ;-)

**Comment by wsavran** [comment, marked as answer]
Thank you for the tip. I'm happy about my first legit customization to my emacs. I'll share what I did in case its useful to anyone else.

edit: added response from below so its correct here.

I defined this function in my `config.el`
```elisp
(defun ws.refs/prettify-author-names (names)
  "Makes list of author names prettier for viewing in citar"
  (let ((short-names (split-string (citar-shorten-names names) ", ")))
    (if (< 2 (length short-names))
        (concat (car short-names) " et al.")
      (string-join short-names " and "))))
```
and I set `citar-display-transform-functions` in my `config.el` like this
```elisp
(setq citar-display-transform-functions
    '((t  . citar-clean-string)
     (("author") . ws.refs/prettify-author-names)))
```

**Reply by bdarcus** [reply]
The first looks good, the second not. You're not defining the variable; you're setting it. Untested, but should be:

```elisp
(setq citar-display-transform-functions
    '((t  . citar-clean-string)
      (("author") . ws.refs/prettify-author-names)))
```

FWIW, the order should probably be swapped; something I need to do when I get to #529.

**Reply by wsavran** [reply]
it works as is. thanks for the feedback.
